app/controllers/import_controller.rb

```diff
@@ -26,5 +34,39 @@ def import
       end
     end
   end
-end
-  
+
+  def import_portal_activity
+    request_json = JSON.parse "#{request.body.read}", :symbolize_names => true
+
+    unless User.find_by_email request_json[:activity][:user_email]
+      user = User.create(
+        email: request_json[:activity][:user_email],
+        password: User.get_random_password,
+        is_author: true
+      )
+    end
+
+    portal = Concord::AuthPortal.portal_for_url(request_json[:portal_url]) 
+    authentication = Authentication.find_by_provider_and_uid(portal.strategy_name,request_json[:domain_uid])
+    user = User.find(authentication.user_id)
```

Comment:
@scytacki Yes, the user is the admin and he is triggering the import. The publishing code already links the author of the activity in LARA in the portal here https://github.com/concord-consortium/rigse/blob/import_itsi_schools_district_users/lib/activity_runtime_api.rb#L55 , in addition to this we have given the author role to this user(author of the activity and not the importer) in the portal here https://github.com/concord-consortium/rigse/blob/import_itsi_schools_district_users/lib/import_external_activity.rb#L29.